Packs/PAN-OS/Playbooks/PAN-OS_-_Configure_DNS_Sinkhole.yml

```diff
@@ -0,0 +1,1963 @@
+  playbookInputQuery:
+- key: AutoCreateRuleToAllowDNS
+  value:
+    simple: "False"
+  required: false
+  description: |-
+    Determines whether the playbook can automatically create a rule that specifically allows DNS traffic from the internal DNS servers to the public DNS server configured in the playbook inputs.
+    Possible values and their meanings are as follows:
+    True - create such a rule automatically if one does not exist (Warning - this will allow previously unallowed traffic).
+    False - do not create such a rule automatically.
+
+    If a value of "False" is specified, the playbook will check if the NetworkAdminEmail input is specified. If it is, the playbook will seek approval to create the rule, from the email specified in that input. If no email is specified, the playbook will abort without making any changes.
+  playbookInputQuery:
+- key: NetworkAdminEmail
+  value: {}
+  required: false
+  description: |-
+    Optional - the email of the person who should approve the creation of a new security policy rule in the firewall. If no rule allows internal DNS traffic to go outside, the playbook will send an email to the email specified in this input, asking to approve the creation of such rule.
+    If no email is specified, and the AutoCreateRuleToAllowDNS input is set to False, the playbook will abort without making any changes, as such a rule is mandatory for the flow of the playbook.
+  playbookInputQuery:
+- key: RuleNameToDenyToSinkhole
+  value:
+    simple: Deny To Sinkhole
+  required: false
+  description: |-
+    The name of the rule that will be created  to deny traffic to the sinkhole address. This is required so that traffic logs will reveal the endpoints attempting to connect to their C&C (which by the end of this playbook will be sinkholed).
+    If a rule with the specified name does not exist, one will be created. The new rule will be placed before rules that allow DNS traffic, as recommended by the PAN-OS best practices.
+    If no value is specified, and a rule for denying the traffic to the sinkhole does not exist, one with the name "Deny To Sinkhole" will be created.
+  playbookInputQuery:
+- key: OverwriteExistingProfile
+  value:
+    simple: "False"
+  required: true
+  description: |-
+    Whether to overwrite an existing anti-spyware security profile if one is currently applied to the rule that allows DNS traffic.
+
+    If set to True - if the rule already has a different anti-spyware profile applied to it, it will be overwritten with the one specified in the SecurityProfileName input.
+
+    If set to False - if the rule already has a different anti-spyware profile applied to it, the playbook will use that profile when sinkholing domains from the specified DNS Signature Source.
+
+    If you are not sure whether to overwrite it or not - leave this as False, since an existing security profile may have additional logic configured to it.
+  playbookInputQuery:
+outputs: []
```

Comment:
No outputs for this playbook?

Reply:
No need as this is just a configuration playbook. If configuration fails it won't complete. The playbook that fetches IPs from the logs has outputs so it can be used if there's ever a need to verify that the configuration worked